contracts/Cornichon.vy

```diff
@@ -23,21 +23,43 @@ name: public(String[64])
 symbol: public(String[32])
 decimals: public(uint256)
 balanceOf: public(HashMap[address, uint256])
+nonces: public(HashMap[address, uint256])
 allowances: HashMap[address, HashMap[address, uint256]]
+version: public(String[32])
 total_supply: uint256
 dai: ERC20
+DOMAIN_SEPARATOR: public(bytes32)
+DOMAIN_TYPE_HASH: constant(bytes32) = keccak256('EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)')
+PERMIT_TYPE_HASH: constant(bytes32) = keccak256("Permit(address owner,address spender,uint256 amount,uint256 nonce,uint256 expiry)")
 
 
 @external
 def __init__(_name: String[64], _symbol: String[32], _supply: uint256):
     self.name = _name
     self.symbol = _symbol
     self.decimals = 18
+    self.version = "1"
     self.dai = ERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F)
     self.balanceOf[msg.sender] = _supply
     self.total_supply = _supply
     log Transfer(ZERO_ADDRESS, msg.sender, _supply)
 
+    self.DOMAIN_SEPARATOR = keccak256(
+        concat(
+            DOMAIN_TYPE_HASH,
+            keccak256(convert(self.name, Bytes[64])),
+            keccak256(convert(self.version, Bytes[32])),
+            convert(chain.id, bytes32),
+            convert(self, bytes32)
+        )
+    )
+
+
+@view
+@external
+def dom_sep() -> uint256:
+    return chain.id
```

Comment:
Is this necessary? `eth_chainId` is an RPC method that's available.

Reply:
Left this form testing, removed now